### PR TITLE
Merge Anaconda recipe

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

**Comment by isuruf** [comment]
@conda-forge-admin, please rerender

**Comment by isuruf** [comment]
Ready for review

File: recipe/meta.yaml [hunk 1]
```diff
@@ -13,22 +13,23 @@ source:
     - ncurses-5.9-gcc-5.patch
 
 build:
-  skip: true  # [win]
-  number: 0
-  detect_binary_files_with_prefix: true
+  skip: True  # [win]
+  number: 3
```
Comment (jakirkham) [inline recipe/meta.yaml]: Should this be `1`?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -95,21 +96,27 @@ test:
...
-    - pkg-config tinfow --libs | grep "^-L${PREFIX}/lib .*-ltinfow$"
-
-    # Usage tests
-    # - bash run_test.sh    # [osx]
-    # Python test will not pass until it is recompiled against ncurses 6.1
-    # - python run_test.py  # [osx]
+    # - pkg-config ncursesw --libs | grep "^-L${PREFIX}/lib -lncursesw -ltinfow$"
+    # - pkg-config tinfow --libs | grep "^-L${PREFIX}/lib -ltinfow$"
+    # - pkg-config ncurses --libs | grep "^-L${PREFIX}/lib -lncurses -ltinfo$"
+    # - pkg-config tinfo --libs | grep "^-L${PREFIX}/lib -ltinfo$"
```
Comment (jakirkham) [inline recipe/meta.yaml]: Why are we dropping the `pkg-config` tests? These were added to satisfy Bioconda's needs. :/

File: recipe/build.sh [hunk 3]
```diff
@@ -1,43 +1,91 @@
...
+# 2. Any `*--as-needed*` is found in EXTRA_LDFLAGS
+# The build system takes care that any `-Wl,-rpath,` in LDFLAGS gets copied into EXTRA_LDFLAGS
+# (and also that any -L${PREFIX}/lib gets translated to -Wl,-rpath,${PREFIX}/lib)
+# the same is not true of -Wl,--as-needed (which is referenced only in gen-pkgconfig.in).
+# One option to fix this is to pass our LDFLAGS as EXTRA_LDFLAGS however this ends up copying across
+# all of our linker flags into the .pc file which means they are forced upon all pkg-config based
+# consumers of ncurses and that is a very bad thing indeed. If we wanted to do that we would:
+# export EXTRA_LDFLAGS=${LDFLAGS}
+# export LDFLAGS=
+# .. but instead it is better to strip off all '-Wl,-rpath,*' and '-L${PREFIX}' from LDFLAGS.
```
Comment (jakirkham) [inline recipe/build.sh]: While I understand procedurally what is happening, am a bit mystified as to why it is happening. Any info about why this is needed?

File: recipe/build.sh [hunk 4]
```diff
@@ -1,43 +1,91 @@
...
+  --build=${BUILD} \
+  --host=${HOST} \
+  --without-debug \
+  --without-ada \
+  --without-manpages \
+  --with-shared \
+  --disable-overwrite \
+  --enable-symlinks \
+  --enable-termcap \
+  --with-pkg-config-libdir="${PREFIX}"/lib/pkgconfig \
```
Comment (jakirkham) [inline recipe/build.sh]: Shouldn't we quote the whole path here?

File: recipe/build.sh [hunk 5]
```diff
@@ -1,43 +1,91 @@
...
+  --enable-pc-files \
+  --with-termlib \
+  --enable-widec
```
Comment (jakirkham) [inline recipe/build.sh]: Am concerned that not providing the non-widec (normal) libraries may break downstream users. Is there a reason not to provide them?

File: recipe/meta.yaml [hunk 6]
```diff
@@ -13,22 +13,23 @@ source:
...
+    - {{ compiler('cxx') }}
     - pkg-config
 
 test:
-  requires:
-    - pkg-config
-  # files:
-    # - run_test.sh         # [osx]
-    # Python test will not pass until it is recompiled against ncurses 6.1
-    # - run_test.py         # [osx]
```
Comment (jakirkham) [inline recipe/meta.yaml]: Understand that these have been a bit troublesome (hence commented). However am disappointed to see not only they are being dropped, but no significant testing of `ncuruses` is replacing them. Can we add some tests beyond trivial file tests? This library has a huge impact on user experience in the terminal and given how important this is would expect to see some testing that takes this into account.

**Comment by jakirkham** [comment]
Mainly concerned about the lack of testing that comes with these rather sweeping changes. Admittedly we have struggled to test this library in recent releases in conda-forge. However that hardly justifies dropping the remaining working tests (`pkg-config`) or simplify giving up on any kind of shell based test. Would like to see the testing here improve before this is merged.

**Comment by ocefpaf** [comment]
@isuruf apologies but I merged #46 before this one :-(
If you don't have time for a rebase please let me know and I'll take care of it.

**Comment by fhoehle** [comment]
@isuruf some parts were introduced into master. Can this PR be closed, or which parts should be kept?

**Pull request closed without merging**